src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -720,7 +720,7 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
                     }
 
                     // open connection based on request details
-                    result = await Instance.Connect(connectParams);
```

Comment:
It could cause problems if unit testing any ConnectionService that's not the static instance. Right now, you can do this but this method will call into the static instance and register connections. That's a bug and also just bad practice (it should be calling its own Connect method).